osu.Game/Overlays/Mods/ModButton.cs

```diff
@@ -64,41 +64,45 @@ private bool changeSelectedIndex(int newIndex)
             if (newIndex >= 0 && !Mods[newIndex].HasImplementation)
                 return false;
 
-            selectedIndex = newIndex;
-            Mod modAfter = SelectedMod ?? Mods[0];
-
-            if (beforeSelected != Selected)
+            Schedule(() =>
             {
-                iconsContainer.RotateTo(Selected ? 5f : 0f, 300, Easing.OutElastic);
-                iconsContainer.ScaleTo(Selected ? 1.1f : 1f, 300, Easing.OutElastic);
-            }
+                selectedIndex = newIndex;
+                Mod modAfter = SelectedMod ?? Mods[0];
 
-            if (modBefore != modAfter)
-            {
-                const float rotate_angle = 16;
+                if (beforeSelected != Selected)
+                {
+                    iconsContainer.RotateTo(Selected ? 5f : 0f, 300, Easing.OutElastic);
+                    iconsContainer.ScaleTo(Selected ? 1.1f : 1f, 300, Easing.OutElastic);
+                }
+
+                if (modBefore != modAfter)
+                {
+                    const float rotate_angle = 16;
 
-                foregroundIcon.RotateTo(rotate_angle * direction, mod_switch_duration, mod_switch_easing);
-                backgroundIcon.RotateTo(-rotate_angle * direction, mod_switch_duration, mod_switch_easing);
+                    foregroundIcon.RotateTo(rotate_angle * direction, mod_switch_duration, mod_switch_easing);
+                    backgroundIcon.RotateTo(-rotate_angle * direction, mod_switch_duration, mod_switch_easing);
 
-                backgroundIcon.Mod = modAfter;
+                    backgroundIcon.Mod = modAfter;
 
-                using (BeginDelayedSequence(mod_switch_duration, true))
-                {
-                    foregroundIcon
-                        .RotateTo(-rotate_angle * direction)
-                        .RotateTo(0f, mod_switch_duration, mod_switch_easing);
+                    using (BeginDelayedSequence(mod_switch_duration, true))
+                    {
+                        foregroundIcon
+                            .RotateTo(-rotate_angle * direction)
+                            .RotateTo(0f, mod_switch_duration, mod_switch_easing);
 
-                    backgroundIcon
-                        .RotateTo(rotate_angle * direction)
-                        .RotateTo(0f, mod_switch_duration, mod_switch_easing);
+                        backgroundIcon
+                            .RotateTo(rotate_angle * direction)
+                            .RotateTo(0f, mod_switch_duration, mod_switch_easing);
 
-                    Schedule(() => displayMod(modAfter));
+                        Schedule(() => displayMod(modAfter));
```

Comment:
Have you maybe looked whether this schedule can be possibly removed? Always a little skeeved out with nested schedules like that.

I attempted to find out what this was for in blame but got nothing (doesn't help that this code is ages old).

Reply:
this is being used to run the function after the `BeginDelayedSequence` specification above.